SKA-ScienceDataProcessor/integration-prototype
# UKSW-10 Unit tests for visibility receive (continued)

<!—UKSW-10 Unit tests for visibility receive (continued) -->

## Description:
A test developed to confirm the behaviour of handling SPEAD heaps by generating in memory SPEAD heaps and calling the function that decodes the header + payload from the memory buffer. 

Note. Changes have been made to stream.c file for visibility baseline count (line 149). When testing it was found that num_baselines is getting converted using be32toh. We don’t see a reason for this conversion as the items are already getting converted using be64toh in line 96. It is the same for visibility time fraction and visibility timestamp count but has not those are not updated.

## Testing instructions:
Build Instructions
To build the code on a local machine, ensure make and CMake are both installed and give the following commands from the current directory:
```
mkdir build
cd build
cmake ..
make
```
To run the unit tests from the build directory, either run
`ctest`
or run the unit test binary directly using:
`./test/recv_test`

## Types of changes


## Checklist:
